clang/test/dpct/helper_files_ref/include/lapack_utils.hpp

```diff
@@ -353,6 +354,752 @@ inline int potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, int n,
   return 0;
 #endif
 }
+
+namespace detail {
+template <typename functor_t, typename... args_t>
+inline int lapack_shim(sycl::queue &q, library_data_t a_type, int *info,
+                       const std::string &lapack_api_name, functor_t functor,
+                       args_t... args) {
+  try {
+    switch (a_type) {
+    case library_data_t::real_float: {
+      functor.template operator()<float>(args...);
+      break;
+    }
+    case library_data_t::real_double: {
+      functor.template operator()<double>(args...);
+      break;
+    }
+    case library_data_t::complex_float: {
+      functor.template operator()<std::complex<float>>(args...);
+      break;
+    }
+    case library_data_t::complex_double: {
+      functor.template operator()<std::complex<double>>(args...);
+      break;
+    }
+    default:
+      throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
+                            "the data type is unsupported");
+    }
+  } catch (oneapi::mkl::lapack::exception const &e) {
+    std::cerr << "Unexpected exception caught during call to LAPACK API: "
+              << lapack_api_name << std::endl
+              << "reason: " << e.what() << std::endl
+              << "info: " << e.info() << std::endl
+              << "detail: " << e.detail() << std::endl;
+    int info_val = static_cast<int>(e.info());
+    if (info)
+      dpct::detail::dpct_memcpy(q, info, &info_val, sizeof(int),
+                                memcpy_direction::host_to_device)
+          .wait();
+    return 1;
+  } catch (sycl::exception const &e) {
+    std::cerr << "Caught synchronous SYCL exception:" << std::endl
+              << "reason: " << e.what() << std::endl;
+    if (info)
+      dpct::detail::dpct_memset(q, info, 0, sizeof(int)).wait();
+    return 1;
+  }
+  return 0;
```

Comment:
set info value

Reply:
the info value (normal case) is set in *_impl functions, this shim function only handles error cases.

Reply:
we can move set info value into lapack_shim, it contains all same logic